rockin/plugins/conveyor_belt/conveyor_belt_plugin.cpp

```diff
@@ -36,7 +36,25 @@ class ConveyorBeltPlugin: public fawkes::Plugin
         ConveyorBeltPlugin(Configuration *config) :
                 Plugin(config)
         {
-            thread_list.push_back(new ConveyorBeltThread());
+            std::string mode = "";
+            if (config->exists("/llsfrb/conveyor-belt/mode")) {
+                std::string config_mode = config->get_string("/llsfrb/conveyor-belt/mode");
+                if (config_mode == std::string("real"))
+                {
+                    mode = "real";
```

Comment:
I thought our discussion was to have different implementations of the ConveyorBeltThread? Then you would just instantiate the "correct" one here.